arsceneview/src/main/java/com/google/ar/sceneform/ux/TranslationController.kt

```diff
@@ -0,0 +1,260 @@
+                        parentNodeTransformMatrix.decomposeRotation(
+                            parentNodeScale,
+                            parentNodeRotation
+                        )
+                        desiredLocalPosition =
+                            parent.transformationMatrix.transformPoint(desiredLocalPosition)
+                        desiredLocalRotation = Quaternion.multiply(
+                            parentNodeRotation.inverted(),
+                            Preconditions.checkNotNull(desiredLocalRotation)
+                        )
+                    }
+                    desiredLocalRotation = calculateFinalDesiredLocalRotation(
+                        Preconditions.checkNotNull(desiredLocalRotation)
+                    )
+                    lastArHitResult = hit
+                    break
+                }
+            }
+        }
+    }
+
+    override fun onEndTransformation(gesture: DragGesture) {
+        val hitResult = lastArHitResult ?: return
+        if (hitResult.trackable.trackingState == TrackingState.TRACKING) {
+            val anchorNode = anchorNodeOrDie
+            val oldAnchor = anchorNode.anchor
+            oldAnchor?.detach()
+            val newAnchor = hitResult.createAnchor()
+            anchorNode.anchor = newAnchor
+
+// TODO: View if it is usefull
+//      Vector3 worldPosition = getTransformableNode().getWorldPosition();
+//      Quaternion worldRotation = getTransformableNode().getWorldRotation();
+//      Quaternion finalDesiredWorldRotation = worldRotation;
+//
+//      // Since we change the anchor, we need to update the initialForwardInLocal into the new
+//      // coordinate space. Local variable for nullness analysis.
+//      Quaternion desiredLocalRotation = this.desiredLocalRotation;
+//      if (desiredLocalRotation != null) {
+//        getTransformableNode().setQuaternion(desiredLocalRotation);
+//        finalDesiredWorldRotation = getTransformableNode().getWorldRotation();
+//      }
+//
+//      anchorNode.setAnchor(newAnchor);
+//
+//      // Temporarily set the node to the final world rotation so that we can accurately
+//      // determine the initialForwardInLocal in the new coordinate space.
+//      getTransformableNode().setWorldRotation(finalDesiredWorldRotation);
+//      Vector3 initialForwardInWorld = getTransformableNode().getForward();
+//      initialForwardInLocal.set(anchorNode.worldToLocalDirection(initialForwardInWorld));
+//
+//      getTransformableNode().setWorldRotation(worldRotation);
+//      getTransformableNode().setWorldPosition(worldPosition);
+        }
+        desiredLocalPosition = Vector3.zero()
+        desiredLocalRotation = calculateFinalDesiredLocalRotation(Quaternion.identity())
+    }
+
+    private val anchorNodeOrDie: ArNode
+        get() {
+            val parent = transformableNode.parent
+            check(parent is ArNode) { "TransformableNode must have an ArNode as a parent." }
```

Comment:
@ThomasGorisse I'll need some help here: As we don't have an ARNode parent here but the node itself is an ARNode we could return transformableNode here. But then the size of the model changes. Could you take a look?